components/sync15/src/engine/bridged_engine.rs

```diff
@@ -87,8 +90,118 @@ pub trait BridgedEngine: Send + Sync {
     fn wipe(&self) -> Result<()>;
 }
 
-// TODO: We should replace this with OutgoingChangeset to reduce the number
-// of types engines need to deal with.
+// This is an adaptor trait - the idea is that engines can implement this
+// trait along with SyncEngine and get a BridgedEngine for free. It's temporary
+// so we can land this trait without needing to update desktop.
+// Longer term, we should remove both this trait and BridgedEngine entirely, sucking up
+// the breaking change for desktop. The main blocker to this is moving desktop away
+// from the explicit timestamp handling and moving closer to the `get_collection_request`
+// model.
```

Comment:
My initial impression is that I like this trait and the underlying plan.  It's also nice that it shows the difference between the two traits.